Source/monster.cpp

```diff
@@ -2098,44 +2098,49 @@ bool MonsterIsTileClear(int i, Point position)
 			return false;
 	}
 
-	return MonsterIsTileSafe(i, position);
+	return MonsterIsTileSafe(Monsters[i], position);
 }
 
 /**
- * @brief Check that the amonster that can open doors can stand on the tile
+ * @brief If a monster can access the given tile (possibly by opening a door)
  */
 bool MonsterIsTileAccessible(int i, Point position)
 {
-	bool isdoor = false;
+	MonsterStruct *monster;
+	if (i < 0)

```

Comment:
```suggestion
	if (i >= 0)
```
Otherwise this is forcing OOB reads. Even better would be to update `MonsterIsTileAccessible` to take a `const MonsterStruct &` 😀(that does require updating the pathfinding code so not suggesting it has to be done now.)

Reply:
Yeah, path finding can also be done for players and nothing, so we would need to adjust for that as well.